<!-- 
 * mitya_teleop.launch
 * Copyright (c) 2017, Robot Mitya.
 * All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 * 
 *     * Redistributions of source code must retain the above copyright
 *       notice, this list of conditions and the following disclaimer.
 *     * Redistributions in binary form must reproduce the above copyright
 *       notice, this list of conditions and the following disclaimer in the
 *       documentation and/or other materials provided with the distribution.
 *     * Neither the name of the Robot Mitya nor the names of its
 *       contributors may be used to endorse or promote products derived from
 *       this software without specific prior written permission.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
 * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
 * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
 * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
 * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
 * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 *
 *  Created on: Nov 26, 2017
 *      Author: Dmitry Dzakhov
 -->

<launch>
	<group ns="robot_mitya">
		<!-- joy node -->
		<node respawn="true" pkg="joy"
		      type="joy_node" name="mitya_joy" >
			<param name="dev" type="string" value="/dev/input/js0" />
			<param name="deadzone" value="0.12" />
		</node>
	
		<group>
		    <param name="head_horizontal_min_degree" type="double" value="-120.0" />
		    <param name="head_horizontal_center_degree" type="double" value="0.0" />
		    <param name="head_horizontal_max_degree" type="double" value="120.0" />
		    <param name="head_vertical_min_degree" type="double" value="-120.0" />
		    <param name="head_vertical_center_degree" type="double" value="-15.0" />
		    <param name="head_vertical_max_degree" type="double" value="10.0" />
		
			<param name="test" type="string" value="Value Common" />
	
			<!-- Axes -->
			<node pkg="mitya_teleop" type="joystick_node" name="joystick">
				<param name="reboot_button" type="int" value="8" />

				<param name="drive_axis_x" type="int" value="3" />
				<param name="drive_axis_y" type="int" value="4" />
				<param name="drive_boost" type="int" value="2" />
				<param name="drive_boost_min_factor" type="double" value="0.25" />
				<param name="drive_boost_max_factor" type="double" value="1.0" />
				<param name="drive_max_value" type="int" value="100" />
				<param name="drive_invert_x" type="bool" value="true" />
				<param name="drive_invert_y" type="bool" value="false" />
				
				<param name="head_mode_button" type="int" value="9" />
				<param name="head_axis_x" type="int" value="0" />
				<param name="head_axis_y" type="int" value="1" />
				<param name="head_move_horizontal_axis" type="int" value="6" />
				<param name="head_move_vertical_axis" type="int" value="7" />
				<param name="head_invert_horizontal" type="bool" value="false" />
				<param name="head_invert_vertical" type="bool" value="false" />
				<param name="head_move_center_button" type="int" value="4" />
				
				<param name="led1_button" type="int" value="3" />
				<param name="led2_button" type="int" value="1" />
				<param name="tail_button" type="int" value="2" />
			</node>
		  
			<!-- Head node -->
			<node pkg="mitya_teleop" type="herkulex_node" name="herkulex">
				<param name="serial_port" type="string" value="/dev/ttyUSB0" />
				<param name="baud_rate" type="int" value="115200" />
			</node>
		</group>
	</group>
</launch>
